ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +265,69 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+//update the description of secret
+const updateSecretDesc = async (res, appid, payload, DB) => {
+    //get and check param in payload
+    let secretName = getParam_String(payload, 'secretName')
+    let description = getParam_String(payload, 'description')
+    if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+        res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+        return
+    }
+    if (!checkStringParam(description, false, DESCRIPTION_LENGTH_MAX)) {
+        res.send(_result(400, `description must be string and length not more than ${DESCRIPTION_LENGTH_MAX}`))
+        return
+    }
+
+    //Query the description through secret name and update the description
+    const secret_name_query = {
+        selector: {
+            appid,
+            secretName
+        },
+        fields: [
+            '_id',
+            '_rev',
+            'appid',
+            'secretName',
+            'encryptionKeyId',
+            'description',
+            'createTime',
+            'deleteTime',
+            'plannedDeleteTime',
+            'rotationInterval',
+            'lastRotationDate',
+            'nextRotationDate'
```

Comment:
does it means all the fields will be returned?

Reply:
yes, need to re insert the fields of the Saved doc